---
title: "Example ChIP-seq analysis"
author: "Pierre-Luc"
date: "3/10/2022"
output: html_document
---

```{r}
suppressPackageStartupMessages({
  library(AnnotationHub)
  library(Rsubread)
  library(rtracklayer)
  library(Biostrings)
  library(Rfastp)
  library(epiwraps)
})
ah <- AnnotationHub()
```

We will perform the primary analysis of an example dataset, at each step demonstrating the use of both a very standard pipeline, and a full R-based equivalent. Pay attention to which chunks are R or bash code (see header)

# Downloading the raw reads

We download an example dataset which is a ChIP-seq experiment mapping the bindings
of the factor Myc across the Kc167 Drosophila cell line.
We download the data from ENCODE:

```{r, eval=FALSE}
download.file("https://www.encodeproject.org/files/ENCFF014MOQ/@@download/ENCFF014MOQ.fastq.gz", dest="raw/Myc.fastq.gz")
```

# Reads QC and trimming

## Using fastqc 

For this we will use shell commands:

```{bash engine.opts='-l', eval=FALSE}
mkdir -p raw_fastqc
fastqc -o raw_fastqc --threads 4 raw/Myc.fastq.gz
```

This will create an html report with QC plots

### Trimming

```{bash engine.opts='-l', eval=FALSE}
mkdir -p trimmed
trimmomatic SE -threads 4 -summary trimmed/Myc.stats -phred33 raw/Myc.fastq.gz trimmed/Myc.fastq.gz ILLUMINACLIP:TruSeq3-SE:2:15:4:4:true LEADING:20 TRAILING:20 SLIDINGWINDOW:4:15 MINLEN:25
```

## Using Rfastp

```{r}
dir.create("rfastp.trimmed")
qc <- Rfastp::rfastp("raw/Myc.fastq.gz", outputFastq="rfastp.trimmed/Myc", 
                     maxNfilter=0, thread=4, overrepresentationAnalysis=TRUE)
```

This will create an html report with before/after QC plots, as well as a trimmed
and filtered fastq file. Furthermore, the R output object (equivalent to the .json 
stats also saved by the program) can be used to generate the QC plots inside a
markdown, for example:

```{r}
Rfastp::curvePlot(qc, curve="content_curves")
```


# Alignment

## Using Rsubread

### Building a genome index for mapping

This we have to do only once for a genome, and can then re-use across projects

```{r, eval=FALSE}
# we get the genome sequence from AnnotationHub
genome <- ah[["AH49674"]]
# we create a new directory that will contain the genome index
dir.create("BDGP6_genome")
# we write the genome sequence in fasta format
export(import.2bit(genome), "BDGP6_genome/genome.fasta.gz", compress=TRUE)
# we build a Rsubread index
Rsubread::buildindex("BDGP6_genome/rsubread", reference="BDGP6_genome/genome.fasta.gz")
```

### Alignment

```{r}
dir.create("aligned")
align.stats <- Rsubread::align(index="BDGP6_genome/rsubread", type="dna",
                               output_file="aligned/Myc.bam",
                               readfile1="rfastp.trimmed/Myc_R1.fastq.gz", 
                               nthreads=4, sortReadsByCoordinates=TRUE)
align.stats
```


## Using bowtie2

### Building a bowtie2 index

```{bash engine.opts='-l'}
# bowtie2-build [options] path/to/genome.fasta path/to/outputIndex
bowtie2-build --threads 4 BDGP6_genome/genome.fasta.gz BDGP6_genome/bowtie2
```

### Alignment

```{bash engine.opts='-l'}
mkdir -p aligned
(bowtie2 -p 4 -x BDGP6_genome/bowtie2 -U trimmed/Myc.fastq.gz) 2> aligned/Myc.bowtie2 |\
  samtools view -bS - | samtools sort -@4 -m 2G - > aligned/Myc.bam
samtools index aligned/Myc.bam
```

# Peak calling

## Using MACS2

```{bash engine.opts='-l'}
mkdir -p peaks
macs2 callpeak --outdir peaks -n Myc --gsize dm -t aligned/Myc.bam
```

We can then import the peaks in R:

```{r}
peaks <- rtracklayer::import.bed("peaks/Myc_peaks.narrowPeak")
```


## Using R

```{r}
peaks <- callPeaks("aligned/Myc.bam")
# if we want to save it as a bed file:
rtracklayer::export.bed(peaks, "peaks/peaks.bed")
```


# Generating coverage tracks

## Using DeepTools

```{bash engine.opts='-l'}
mkdir -p tracks
bamCoverage -p 2 --binSize 20 --normalizeUsing CPM -b aligned/Myc.bam -o tracks/Myc.bw
```


## Using R

```{r}
dir.create("tracks")
bam2bw("aligned/Myc.bam", "tracks/Myc_full.bw", binWidth=1, extend=50, scaling=TRUE)
bam2bw("aligned/Myc.bam", "tracks/Myc.bw", binWidth=10, extend=50, scaling=TRUE)
```

# Looking at an example peak

```{r}
plotSignalTracks(list(full="tracks/Myc_full.bw", binned="tracks/Myc.bw", peaks=peaks), region=peaks[1], extend=500)
```


```{r}
plotSignalTracks(list(signal="tracks/Myc.bw", peaks=peaks), region="numb", ensdb=ensdb)
```



# Basic peak annotation

```{r}
ensdb <- ah[["AH95713"]] # Drosophila EnsDb
peakAnno <- annotateRegions(peaks, ensdb)
table(peakAnno$class)
```

