---
title: "CTCF Journey Across Neuronal Development"
author: "Viviana Vinci"
date: "Bioinformatic Approaches to Regulatory Genomics and Epigenomics, ETH ZÃ¼rich. July 2022"
output:
  html_document: 
   toc: true
   toc_float: true
---



```{r, include=FALSE, warning=FALSE}

# Opening All the Packages 
suppressPackageStartupMessages({
  library(AnnotationHub)
  library(Rsubread)
  library(rtracklayer)
  library(Biostrings)
  library(Rfastp)
  library(epiwraps)
  library(GEOquery)
  library(GenomicRanges)
  library(IRanges)
  library(ChIPpeakAnno)
  library(ggplot2)
  library(gridExtra)
  library(ChIPseeker)
  library("TxDb.Mmusculus.UCSC.mm10.knownGene")
  library(org.Mm.eg.db)
  library(cowplot)
  library(ggplot2)
  library(Gviz)

})

```


```{r, include=FALSE}

#Downloading all the tracks and peaks of Cortical Neurons 

tracks_CN_CTCF<- rtracklayer::import('CN/tracks/GSE96107_CN_CTCF.bw')

tracks_CN_H3K27ac<-rtracklayer::import('CN/tracks/GSE96107_CN_H3K27ac.bw')

tracks_CN_H3K9me3<-rtracklayer::import('CN/tracks/GSE96107_CN_H3K9me3.bw')

peaks_CN_CTCF<- rtracklayer::import('CN/peaks/GSE96107_CN_CTCF.IDR0.05.filt.narrowPeak')


peaks_CN_H3K27ac <- rtracklayer::import('CN/peaks/GSE96107_CN_H3K27ac.IDR0.05.filt.narrowPeak')

#download.file("https://www.encodeproject.org/files/ENCFF356LFX/@@download/ENCFF356LFX.bed.gz" , dest="CN/peaks/CN_H3K9me3.bed.gz")
peaks_CN_H3K9me3<- rtracklayer::import('CN/peaks/CN_H3K9me3.bed.gz')

```


```{r, include=FALSE}

#Downloading all the Tracks and Peaks of Neuronal Progenitor Neurons 

tracks_NPC_CTCF<- rtracklayer::import("NPC /tracks/GSE96107_NPC_CTCF.bw")

tracks_NPC_H3K27ac<- rtracklayer::import('NPC /tracks/GSE96107_NPC_H3K27ac.bw')

tracks_NPC_H3K9me3<- rtracklayer::import('NPC /tracks/GSE96107_NPC_H3K9me3.bw')

peaks_NPC_CTCF<- rtracklayer::import('NPC /peaks/GSE96107_NPC_CTCF.IDR0.05.filt.narrowPeak')


peaks_NPC_H3K27ac <- rtracklayer::import('NPC /peaks/GSE96107_NPC_H3K27ac.IDR0.05.filt.narrowPeak')



#download.file("https://www.encodeproject.org/files/ENCFF719LDJ/@@download/ENCFF719LDJ.bed.gz" , dest="NPC /peaks/NPC_H3K9me3.narrowPeak")

peaks_NPC_H3K9me3<- rtracklayer::import('NPC /peaks/NPC_H3K9me3.narrowPeak')

```


```{r, include=FALSE}

tracks_ES_CTCF<- rtracklayer::import("ES/tracks/GSE96107_ES_CTCF.bw")

tracks_ES_H3K27ac<- rtracklayer::import('ES/tracks/ENCFF163HEV.bigWig')

tracks_ES_H3K9me3<- rtracklayer::import('ES/tracks/ENCFF549VBE.bigWig')

peaks_ES_CTCF<- rtracklayer::import('ES/peaks/GSE96107_ES_CTCF.IDR0.05.filt.narrowPeak')


#download.file("https://www.encodeproject.org/files/ENCFF909QOC/@@download/ENCFF909QOC.bed.gz" , dest="ES/peaks/ES_H3K9me3.narrowPeak")

#download.file("https://www.encodeproject.org/files/ENCFF519QMV/@@download/ENCFF519QMV.bed.gz" , dest="ES/peaks/ES_H3K27ac.narrowPeak")

peaks_ES_H3K27ac <- rtracklayer::import('ES/peaks/ES_H3K27ac.narrowPeak')

peaks_ES_H3K9me3<- rtracklayer::import('ES/peaks/ES_H3K9me3.narrowPeak')

```



<style>
#TOC {
 font-family: Calibri;
 font-size: 16px;

}
body {
 font-family: Calibri;
 font-size: 16px;
}

</style>

## **Introduction** 
### *The Genome in 3D and CTCF role in Gene Loop Formation*

In the last decade, we have become more conscious of the complex structure of the genome and we are starting to understand that it is far more complicated and interesting than just a liner sequence of base pairs(1).
It is now clear that genomes have 3D architectures and that their structure includes functional roles for genomic activity(2). Inside the nucleus there are different levels of DNA organization: chromosomes, compartments A and B, Topologically Associated Domains (TADs), gene loops and chromatin fibers. Each of them plays an important structural and functional role. The advancement of chromatin capture technology has revealed how chromatin loops are formed(3). As a result of the data generated by Hi-C maps, it has been noticed that many of the contact points in the genome are characterized by CTCF binding sites and the occupancy of this regulatory protein at those sites(4). Chromatin loops are formed by a loop extrusion process through a ring-like structured protein called cohesin and the loop extrusion process is halted by the boundary protein CTCF. Evidence suggests that loops are important for the enhancer-gene interaction and that changes in loops structure affect gene expression and function(5). Interest has been raised in the structural changes during cellular differentiation; evidence suggests that the changes in structure are involved in fate determination. In this report ChIPSeq datasets from a study conducted from Bonev et al 2017(6) are used to explore the changes in CTCF binding across three stages of neuronal development, along with two histone marks: H3K27ac and H3K9me3. Furthermore, those datasets will be used to investigate CTCF binding around three Immediate Early Genes (IEGs), *Gadd45y*, *Arc* and *Npas4*. Investigating IEGs in neuronal cell lines is important because they have been implicated in learning and memory(7) , they are promptly transcribed after neuronal activation and they have been commonly found inside chromatin loops; suggesting a fundamental role of the loop structure for their readiness to be transcribed. 


## **Results**

### Feature Distribution and Distance to Transcription Start Site of CTCF, H3K27ac, H3K9me3 across Neuronal Development

To gain a better understanding of CTCF binding across the genome, the feature distribution was analyzed in three neuronal developmental stages. 
In all cell types, CTCF showed increased binding in Distal intergenic regions, Introns and at the promoter, while H3K27ac showed (Figure 1). Interestingly, H3K27ac binding at the promotes seems to increase in percentage as the cells become more differentiated, likewise H3K9me3 binding at the promoters increases in percentage across neuronal development (Fig. 1: A, C, E). 
A similar trend is observed when the binding to the proximity of TSS is analyzed. As cells differentiate the binding of H3K27ac and H3K9me3 seems increase very close to TSS (0-1kb) (Fig. 1: B, D, F). 


```{r, warning=FALSE, collapse=TRUE, fig.cap=" **Figure 1:** CTCF, H3K27ac, H3K9me3 Peak Feature Distribution across the genome in Cortical Neurons (**A,B**), Neuronal Progenitor Cells ( **C,D** ), Embryonic Cells ( **E,F** )", out.width="90%", out.height="90%"}

peakAnnoList_CN <- lapply(list(peaks_CN_CTCF, peaks_CN_H3K27ac, peaks_CN_H3K9me3) ,annotatePeak, TxDb=TxDb.Mmusculus.UCSC.mm10.knownGene, tssRegion=c(-3000, 3000), verbose=FALSE)
names(peakAnnoList_CN) <- c("CTCF","H3K27ac", "H3K9me3")

Pab_CN<- plotAnnoBar(peakAnnoList_CN, title= "Cortical Neurons Feature Distribution")

pdtss_CN<- plotDistToTSS(peakAnnoList_CN, title= "Cortical Neurons Distribution of TF binding loci relative to TSS")

peakAnnoList_NPC <- lapply(list(peaks_NPC_CTCF, peaks_NPC_H3K27ac, peaks_NPC_H3K9me3) ,annotatePeak, TxDb=TxDb.Mmusculus.UCSC.mm10.knownGene, tssRegion=c(-3000, 3000), verbose=FALSE)

names(peakAnnoList_NPC) <- c("CTCF","H3K27ac", "H3K9me3")

Pab_NPC<- plotAnnoBar(peakAnnoList_NPC, title= "Neuronal Progenitor Feature Distribution")

pdtss_NPC<-plotDistToTSS(peakAnnoList_NPC, title= "Neuronal Progenitor Cells Distribution of TF binding loci relative to TSS")

peakAnnoList_ES <- lapply(list(peaks_ES_CTCF, peaks_ES_H3K27ac, peaks_ES_H3K9me3) ,annotatePeak, TxDb=TxDb.Mmusculus.UCSC.mm10.knownGene, tssRegion=c(-3000, 3000), verbose=FALSE)
names(peakAnnoList_ES) <- c("CTCF","H3K27ac", "H3K9me3")

Pab_ES<- plotAnnoBar(peakAnnoList_ES, title= "Embryonic Cells Feature Distribution")

pdtss_ES<-plotDistToTSS(peakAnnoList_ES, title = "Embryonic Cells Distribution of TF binding loci relative to TSS")

plot_grid(Pab_CN,pdtss_CN, nrow = 2, ncol = 1, labels = c("A","B"))
plot_grid(Pab_NPC, pdtss_NPC, nrow = 2, ncol = 1, labels = c("C","D"))
plot_grid(Pab_ES, pdtss_ES, nrow = 2, ncol = 1, labels = c("E","F"))

```


### Coverage Tracks Around *Gadd45y* Across Neuronal Development 

In order to keep investigating the relations between CTCF and histone marks H3K27ac and H3K9me3 a closer look at three immediate early genes was taken in all different developmental stages. 
Figure 2 shows the peaks obtained from the ChipSeq analysis in Cortical Neurons(CN) from coordinates 51700000 bp to 51900000 bp of chromosome 13. This representation shows a possible gene loop right outside Gadd45y.  Some of the peaks of CTCF are really close to H3K27ac, a feature that will be discussed in the Discussion session but that might indicate nucleosome displacement and active enhancer sites, that allows CTCF to be localized at that site. 
It is also noted that really proximal to Gadd45y there are many CTCF binding sites that are retained throughout neuronal development. The most prominent peaks are retained from ES stage to CN.
The ChIPSeq performed in Neuronal Progenitor Cells (NPC) (Figure3) indicates a loss in H3K27ac binding sites. H3K9me3 seems not to be present at this location neither in CN nor in NPC. 
Embryonic cells data reveals more peaks in all three factors examined, especially an increase in peaks of H3K9me3. 



```{r, collapse=TRUE, warning=FALSE}


ideogram13<- IdeogramTrack("chr13", "mm10")
axis<- GenomeAxisTrack()

cover_track_CTCF_CN<-DataTrack(tracks_CN_CTCF, window= 100000,type= "h", name="CTCF", col="blue")
cover_track_H3K27ac_CN<-DataTrack(tracks_CN_H3K27ac, window= 100000,type= "h", name="H3K27ac", col="purple")
cover_track_H3K9me3_CN<-DataTrack(tracks_CN_H3K9me3, window= 100000,type= "h", name="H3K9me3", col="pink", ylim = c(0,2) )

peak_track_CTCF_CN<-AnnotationTrack(peaks_CN_CTCF,name="Peaks")
peak_track_H3K27ac_CN<-AnnotationTrack(peaks_CN_H3K27ac,name="Peaks")
peak_track_H3K9me3_CN<-AnnotationTrack(peaks_CN_H3K9me3,name="Peaks")

tx_13<-GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene ,chromosome = "chr13",start  = 51700000, end= 51900000, name="Genes")

tracks_list_CN_13<-list(ideogram13,cover_track_CTCF_CN,peak_track_CTCF_CN, cover_track_H3K27ac_CN, peak_track_H3K27ac_CN, cover_track_H3K9me3_CN,peak_track_H3K9me3_CN, tx_13, axis)
```


```{r, fig.cap="**Figure 2:** ChipSeq around Gadd45y in Cortical Neurons, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}

plotTracks(tracks_list_CN_13,from = 51700000, to= 51900000, chromosome = "chr13",main= "ChipSeq around Gadd45y in cortical Neurons",cex.main=1.5)
```


```{r, collapse=TRUE, warning=FALSE}

cover_track_CTCF_NPC<-DataTrack(tracks_NPC_CTCF, window= 100000,type= "h", name="CTCF", col="light blue", ylim = c(0,5))
cover_track_H3K27ac_NPC<-DataTrack(tracks_NPC_H3K27ac, window= 100000,type= "h", name="H3K27ac", col="purple",ylim = c(0,5))
cover_track_H3K9me3_NPC<-DataTrack(tracks_NPC_H3K9me3, window= 100000,type= "h", name="H3K9me3", col="pink", ylim = c(0,5) )

peak_track_CTCF_NPC<-AnnotationTrack(peaks_NPC_CTCF,name="Peaks")
peak_track_H3K27ac_NPC<-AnnotationTrack(peaks_NPC_H3K27ac,name="Peaks")

peaks_chr13_NPC<- peaks_NPC_H3K9me3[seqnames(peaks_NPC_H3K9me3)=="chr13"]
peak_track_H3K9me3_NPC_13<-AnnotationTrack(peaks_chr13_NPC,name="Peaks")

tracks_list_NPC_13<-list(ideogram13,cover_track_CTCF_NPC,peak_track_CTCF_NPC,cover_track_H3K27ac_NPC,peak_track_H3K27ac_NPC, cover_track_H3K9me3_NPC,peak_track_H3K9me3_NPC_13, tx_13, axis)
```


```{r,fig.cap= "**Figure 3:** ChipSeq around Gadd55y in Neuronal Progenitor Cells, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}
plotTracks(tracks_list_NPC_13,from = 51700000, to= 51900000, chromosome = "chr13",main= "ChipSeq around Gadd45y in Neuronal Progenitor Cells",cex.main=1.5)
```



```{r, include=TRUE, collapse=TRUE, warning=FALSE}

cover_track_CTCF_ES<-DataTrack(tracks_ES_CTCF, window= 100000,type= "h", name="CTCF", col=" dark blue",ylim = c(0,10))
cover_track_H3K27ac_ES<-DataTrack(tracks_ES_H3K27ac, window= 100000,type= "h", name="H3K27ac", col="purple", ylim = c(0,10))
cover_track_H3K9me3_ES<-DataTrack(tracks_ES_H3K9me3, window= 100000,type= "h", name="H3K9me3", col="pink", ylim = c(0,10) )
peaks_chr13<- peaks_ES_CTCF[seqnames(peaks_ES_CTCF)=="chr13"]
peak_track_CTCF_ES_13<-AnnotationTrack(peaks_chr13,name="Peaks")

peaks_chr13_H3K27ac<- peaks_ES_H3K27ac[seqnames(peaks_ES_H3K27ac)=="chr13"]
peak_track_H3K27ac_ES_13<-AnnotationTrack(peaks_chr13_H3K27ac,name="Peaks")

peaks_chr13_H3K9me3<- peaks_ES_H3K9me3[seqnames(peaks_ES_H3K9me3)=="chr13"]
peak_track_H3K9me3_ES_13<-AnnotationTrack(peaks_chr13_H3K9me3,name="Peaks H3K9me3")


tracks_list_ES_13<-list(ideogram13, cover_track_CTCF_ES, peak_track_CTCF_ES_13, cover_track_H3K27ac_ES, peak_track_H3K27ac_ES_13, cover_track_H3K9me3_ES, peak_track_H3K9me3_ES_13, tx_13, axis)
```


```{r, include=TRUE, collapse=TRUE, fig.cap= "**Figure 4:** ChipSeq around Gadd55y in Embryonic Cells, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}
plotTracks(tracks_list_ES_13,from = 51700000, to= 51900000, chromosome = "chr13",main= "ChipSeq around Gadd45y in Embryonic Cells", cex.main=1.5)
```


### Coverage Tracks Around *Arc* Across Neuronal Development 

Similarly, when investigating the binding of CTCF, H3K27ac and H3K9me3 around Arc it appears that as cells develop from the embryonic stage (Figure 7) to cortical neurons (Figure 5), they lose binding sites for all three factors, however the more evident lost is for H3K9me3 bindings. CTCF binding seems to refine as cells develop. H3K27ac seems to disappear during the NPC (Figure 6) cell state but then to appear again at different sites. 



```{r, collapse=TRUE, warning=FALSE}

ideogram15<- IdeogramTrack("chr15", "mm10")

tx_15<-GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene ,chromosome = "chr15",start  = 74600000, end= 74690000, name="Genes")

tracks_list_CN_15<-list(ideogram15,cover_track_CTCF_CN,peak_track_CTCF_CN, cover_track_H3K27ac_CN, peak_track_H3K27ac_CN, cover_track_H3K9me3_CN,peak_track_H3K9me3_CN, tx_15, axis)
```


```{r ChipSeq around Arc in cortical Neurons, fig.cap="**Figure 5:** ChipSeq around Arc in Cortical Neurons, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}

plotTracks(tracks_list_CN_15,from = 74600000, to= 74690000, chromosome = "chr15",main= "ChipSeq around Arc in cortical Neurons",cex.main=1.5)
```


```{r, collapse=TRUE, warning=FALSE}
peaks_NPC_H3K9me3_15<- peaks_NPC_H3K9me3[seqnames(peaks_NPC_H3K9me3)=="chr15"]
peak_track_H3K9me3_NPC_15<-AnnotationTrack(peaks_NPC_H3K9me3_15,name="Peaks")

tracks_list_NPC_15<-list(ideogram15,cover_track_CTCF_NPC,peak_track_CTCF_NPC,cover_track_H3K27ac_NPC,peak_track_H3K27ac_NPC, cover_track_H3K9me3_NPC,peak_track_H3K9me3_NPC_15, tx_15, axis)
```


```{r, collapse=TRUE, warning=FALSE, fig.cap="**Figure 6:** ChipSeq around Arc in Neuronal Progenitor Cells, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}

plotTracks(tracks_list_NPC_15,from = 74600000, to= 74690000, chromosome = "chr15",main= "ChipSeq around Arc in Neuronal Progenitor Cells",cex.main=1.5, ylim= c(0,3))
```


```{r}
peaks_chr15<- peaks_ES_CTCF[seqnames(peaks_ES_CTCF)=="chr15"]
peak_track_CTCF_ES_15<-AnnotationTrack(peaks_chr15,name="Peaks CTCF")

peaks_chr15_H4K27ac<- peaks_ES_H3K27ac[seqnames(peaks_ES_H3K27ac)=="chr15"]
peak_track_H3K27ac_ES_15<-AnnotationTrack(peaks_chr15_H4K27ac,name="Peaks H3K27ac")

peaks_chr15_H3K9me3<- peaks_ES_H3K9me3[seqnames(peaks_ES_H3K9me3)=="chr15"]
peak_track_H3K9me3_ES_15<-AnnotationTrack(peaks_chr15_H3K9me3,name="Peaks H3K9me3")


tracks_list_ES<-list(ideogram15, cover_track_CTCF_ES, peak_track_CTCF_ES_15, cover_track_H3K27ac_ES, peak_track_H3K27ac_ES_15, cover_track_H3K9me3_ES, peak_track_H3K9me3_ES_15, tx_15, axis)

```


```{r ChipSeq around Arc in Embryonic Cells, fig.cap="**Figure 7:** ChipSeq around Arc in Embryonic Cells, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}


plotTracks(tracks_list_ES,from = 74600000, to= 74690000, chromosome = "chr15",main= "ChipSeq around Arc in Embryonic Stem Cells ",cex.main=1.5)
```


### Coverage Tracks Around *Npas4* Across Neuronal Development 

Lastly, the gene Npas4 was analyzed, and similarly the histone mark H3K9me3 disappeared throughout development, with prominent peaks in embryonic stage (Figure 10) but no peaks in NPC and CN (Figure 9 , 8).  H3K27ac is lost from ES to NPC and is gained again from NPC to CN. CTCF peaks seem to be refined as the cell passes to a more defined developmental stage, similarly to all the other IEGs observed in this study. 


```{r, collapse=TRUE, warning=FALSE}

ideogram19<- IdeogramTrack("chr19", "mm10")

tx_19<-GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene ,chromosome = "chr19",start  = 4900000, end= 5004439, name="Genes")

tracks_list_CN<-list(ideogram19,cover_track_CTCF_CN,peak_track_CTCF_CN, cover_track_H3K27ac_CN, peak_track_H3K27ac_CN, cover_track_H3K9me3_CN,peak_track_H3K9me3_CN, tx_19, axis)
```


```{r ChipSeq around Npas4 in Cortical Neurons, fig.cap="**Figure 8:** ChipSeq around Npas4 in Cortical Neurons, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}

plotTracks(tracks_list_CN,from = 4900000, to= 5004439, chromosome = "chr19",main= "ChipSeq around Npas4 in Cortical Neurons",cex.main=1.5, ylim=c(0,4))
```


```{r, collapse=TRUE, warning=FALSE}

peaks_chr19_H3K9me3<- peaks_NPC_H3K9me3[seqnames(peaks_NPC_H3K9me3)=="chr19"]
peak_track_H3K9me3_NPC_19<-AnnotationTrack(peaks_chr19_H3K9me3,name="Peaks H3K9me3")

tracks_list_NPC<-list(ideogram19,cover_track_CTCF_NPC,peak_track_CTCF_NPC,cover_track_H3K27ac_NPC,peak_track_H3K27ac_NPC, cover_track_H3K9me3_NPC,peak_track_H3K9me3_NPC_19, tx_19, axis)


```



```{r, fig.cap="**Figure 9:** ChipSeq around Npas4 in Neuronal Progenitor Cells, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}

plotTracks(tracks_list_NPC,from = 4900000, to= 5004439, chromosome = "chr19",main= "ChipSeq around Npas4 in Neuronal Progenitor Cells",cex.main=1.5, ylim=c(0,6))
```

```{r, collapse=TRUE, warning=FALSE}

peaks_chr19<- peaks_ES_CTCF[seqnames(peaks_ES_CTCF)=="chr19"]
peak_track_CTCF_ES_19<-AnnotationTrack(peaks_chr19,name="Peaks")

peaks_chr19_H3K27ac<- peaks_ES_H3K27ac[seqnames(peaks_ES_H3K27ac)=="chr19"]
peak_track_H3K27ac_ES_19<-AnnotationTrack(peaks_chr19_H3K27ac,name="Peaks")

peaks_chr19_H3K9me3<- peaks_ES_H3K9me3[seqnames(peaks_ES_H3K9me3)=="chr19"]
peak_track_H3K9me3_ES_19<-AnnotationTrack(peaks_chr19_H3K9me3,name="Peaks H3K9me3")

tracks_list_ES<-list(ideogram19, cover_track_CTCF_ES, peak_track_CTCF_ES_19, cover_track_H3K27ac_ES, peak_track_H3K27ac_ES_19, cover_track_H3K9me3_ES, peak_track_H3K9me3_ES_19, tx_19, axis)


```



```{r, fig.cap="**Figure 10:** ChipSeq around Npas4 in Embryonic Cells, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}

plotTracks(tracks_list_ES,from = 4900000, to= 5004439, chromosome = "chr19",main= "ChipSeq around Npas4 in Embryonic Cells ",cex.main=1.5)
```
### A Closer look intro CTCF binding Across Neuronal Development 

To visualize more closely the CTCF peaks across development, the peaks for each developmental stage were plotted using the *plotTracks* function for all three IEGs of relevance (Figure 11, 12, 13). The trend shows similar refinement of CTCF binding decreasing the number of binding sites as the cells differentiate. More speculations regarding this will be mentioned in the Discussion session. 


```{r, fig.cap="**Figure 11**: Coverage tracks of CTCF binding around Gadd45y in Cortical Neurons(blue), Neuronal Progenitor Cells(light blue) and Embryonic Cells(dark blue), peak tracks are depicted showing the respective peaks called for each cell type"}


plotTracks(list(ideogram13, cover_track_CTCF_CN, peak_track_CTCF_CN, cover_track_CTCF_NPC, peak_track_CTCF_NPC, cover_track_CTCF_ES, peak_track_CTCF_ES_13, tx_13, axis),from = 51799965, to= 51859475, chromosome = "chr13",main= "CTCF binding around Gadd45 in ES", cex.main=1.5, legend=TRUE, ylim=c(0,10))


```




```{r, fig.cap="**Figure 12**: Coverage tracks of CTCF binding around Arc in Cortical Neurons(blue), Neuronal Progenitor Cells(light blue) and Embryonic Cells(dark blue), peak tracks are depicted showing the respective peaks called for each cell type"}


plotTracks(list(ideogram15, cover_track_CTCF_CN, peak_track_CTCF_CN, cover_track_CTCF_NPC, peak_track_CTCF_NPC, cover_track_CTCF_ES, peak_track_CTCF_ES_15, tx_15, axis),from = 74600000, to= 74690000, chromosome = "chr15",main= "CTCF binding around Arc throughout Development", cex.main=1.5, ylim=c(0,10))


```



```{r, fig.cap="**Figure 13**: Coverage tracks of CTCF binding around Npas4 in Cortical Neurons(blue), Neuronal Progenitor Cells(light blue) and Embryonic Cells(dark blue), peak tracks are depicted showing the respective peaks called for each cell type"}


plotTracks(list(ideogram19, cover_track_CTCF_CN, peak_track_CTCF_CN, cover_track_CTCF_NPC, peak_track_CTCF_NPC, cover_track_CTCF_ES, peak_track_CTCF_ES_19, tx_19, axis),from = 4900000, to= 5004439, chromosome = "chr19",main= "CTCF binding around Npas4 throughout Development", cex.main=1.5, ylim=c(0,10))


```




## **Methods**
### Data download and importation 

All the data was either manually downloaded if provided by the authors(6) in the GEO repository GSE96107 or it was downloaded from ENCODE data sets, all the path files to download can be found in the initial chuck of the final Rmd or in the supplementary data provided in the final folder. All the data is provided on a folder uploaded to GitHub. Two types of data sets where obtained for each conditions: the coverage tracks in a BigWig format, and the peak tracks in a bed file in a Narrow Peak format  


### ChipSeq Peak and Tracks Formation 

Chromatin immunoprecipitation is a widely used method to identify binding sites of Transcription Factors throughout the genome8. In this study ChIP Seq data is used to look at specific loci in the genome. To do so the coverage tracks are obtained with the function *DataTrack*. And the peaks are annotated using the *AnnotationTrack* function. The genes of the genome are annotated using the function *GeneRegionTrack*. In orther to plot the tracks the *plotTracks* function was used. All four functions are in the *Gviz* package. The information about those functions was obtained at the following website (https://rdrr.io/bioc/Gviz/man/plotTracks.html).  


### Pie Charts 

In order to identify the feature distribution the *annotatePeak* function was used and the *plotAnnotateBar* from the *ChIPSeeker* package was used. This allowed to visualize what was the feature distribution of each condition. 
In order to visualize the distance to TSS a similar function form the same package was used *plotDistToTSS*. The function *plot_grid* from the *cowplot* package was used to merge the plots together.

### Additional Formatting Info 

The entire code was assembled first in separate markdowns that are provided as supplementary information, and then merged into a final report. All the material necessary for the report is found in the final Rmd or HTML, however if something else might be missing, the additional files are provided for further support.  
The formatting style was obtained by studying the information in the following link (https://bookdown.org/yihui/rmarkdown/html-document.html). Many chucks are *include=FALSE* to provide a better final format, but they can be consulted in the final Rmd file. 


## **Discussion**
## CTCF changes across Neuronal Development in Immediate Early Genes 

It has become of real importance to understand and study the epigenome. As our understanding about the complex systems of gene expression increases it is clear how dynamic the system is. Studying the ways, the epigenome evolves in time and through external stimuli is of extreme importance in understanding cellular processes like cellular differentiation. 
CTCF has gained a lot of interest in the scientific community, as it plays a primary role in the 3D organization of the genome(9). In this study the role of CTCF is evaluated throughout three neuronal cell differentiation stages. 
It can be observed in Figure 1, the binding of CTCF seems to be more proximal to the TSS in Cortical Neurons. More prominent trends can be observed for H3K27ac and H3K9me3. For H3K27ac the percentage of peaks at the promoter increases in CN compared to ES, and the bindings seem to be more proximal to the transcription start site. H3K9me3 binding at the promoter seems to also increase in further developed cell types, but it is still mainly found far away from the TSS. H3K27ac is associated with open chromatin states and often associated with enhancer activity(10). Meanwhile, H3K9me3 is associated with repressed chromatin and a heterochromatin conformation. It must be kept in mind that those depicted are the percentages of the overall peaks found for each category and that the number of peaks for ES was much higher than in CN (for example for CTCF there were 121406 peaks in ES compared to 28146 in CN). Even though those results alone are not conclusive, because no statistical analysis has been performed, the trend observed suggests a refinement of CTCF binding after neuronal differentiation and maybe even though the peaks decrease, they are more refined to specific loci of activity. 
As mentioned in the Results section, the H3K9me3 peaks seem to be lost in the IEGs of interest after neuronal differentiation. This result makes sense since in Cortical Neurons those genes must be promptly ready for transcription. In all three of the genes the H3K27ac peaks seem to be lost or partially lost in NPC and then reappear in CN. This could be explained by more accessibility of possible enhancer regions in a state where the neuron has to be ready to promptly transcribe the IEGs. Moreover, many of the H3K27ac peaks are proximal to the CTCF binding sites, this might suggest a nucleosome displacement trend that allows CTCF to properly bind to its site. CTCF is a fundamental protein that has been shown to be important for long-distance enhancer-promoter interaction(11), it is still yet to be understood whether CTCF plays a fundamental role in the activation of those genes. *Gadd45y* has fundamental roles is all systems of the body, but it has been classified as an immediate early gene because its transcription rapidly increases after neuronal stimulation. *Arc* is a master regulator in synaptic plasticity and has been known to play a role in LTD and LTP. *Npas4* is a transcription factor involved in the regulation of the genes involved in excitatory and inhibitory imbalance. It also plays a role in plasticity and circuit formation(12). Therefor it is extremely important to continue investigating how the 3D structure of the genome affects the expression of IEGs. Further studies are required to elucidate the exact mechanisms of gene loop dynamics in the context of immediate early genes. Further studies might include performing similar experiments with animal tissue instead of *in-vitro* cell culture and right after neuronal activation. 


```{r, include=FALSE}
sessionInfo()
```

## **References**

1.	Misteli, T. Beyond the Sequence: Cellular Organization of Genome Function. Cell 128, 787-800, doi:10.1016/j.cell.2007.01.028 (2007).

2.	Misteli, T. The Self-Organizing Genome: Principles of Genome Architecture and Function. Cell 183, 28-45, doi:https://doi.org/10.1016/j.cell.2020.09.014 (2020).

3.	Akgol Oksuz, B. et al. Systematic evaluation of chromosome conformation capture assays. Nature Methods 18, 1046-1055, doi:10.1038/s41592-021-01248-7 (2021).

4. Fudenberg, G. et al. Formation of Chromosomal Domains by Loop Extrusion. Cell Reports 15, 2038-2049, doi:10.1016/j.celrep.2016.04.085 (2016).

5.	Greenwald, W. W. et al. Subtle changes in chromatin loop contact propensity are associated with differential gene regulation and expression. Nature Communications 10, 1054, doi:10.1038/s41467-019-08940-5 (2019).

6.	Bonev, B. et al. Multiscale 3D Genome Rewiring during Mouse Neural Development. Cell 171, 557-572.e524, doi:10.1016/j.cell.2017.09.043 (2017).

7.	Marballi, K. K. & Gallitano, A. L. Immediate Early Genes Anchor a Biological Pathway of Proteins Required for Memory Formation, Long-Term Depression and Risk for Schizophrenia. Front Behav Neurosci 12, 23-23, doi:10.3389/fnbeh.2018.00023 (2018).

8.	Schmidl, C., Rendeiro, A. F., Sheffield, N. C. & Bock, C. ChIPmentation: fast, robust, low-input ChIP-seq for histones and transcription factors. Nature Methods 12, 963-965, doi:10.1038/nmeth.3542 (2015).

9. 	Phillips, J. E. & Corces, V. G. CTCF: Master Weaver of the Genome. Cell 137, 1194-1211, doi:https://doi.org/10.1016/j.cell.2009.06.001 (2009).

10. 	Lewis, E. M. A. et al. Epigenetic regulation during human cortical development: Seq-ing answers from the brain to the organoid. Neurochem Int 147, 105039-105039, doi:10.1016/j.neuint.2021.105039 (2021).


11.  Kubo, N. et al. Promoter-proximal CTCF binding promotes distal enhancer-dependent gene activation. Nat Struct Mol Biol 28, 152-161, doi:10.1038/s41594-020-00539-5 (2021).

12. 	Fu, J., Guo, O., Zhen, Z. & Zhen, J. Essential Functions of the Transcription Factor Npas4 in Neural Circuit Development, Plasticity, and Diseases. Frontiers in Neuroscience 14, doi:10.3389/fnins.2020.603373 (2020).





