---
title: "ES_Arc"
author: "Viviana"
date: '2022-07-02'
output: html_document
editor_options: 
  chunk_output_type: console
---



```{r, include=FALSE}
suppressPackageStartupMessages({
  library(AnnotationHub)
  library(Rsubread)
  library(rtracklayer)
  library(Biostrings)
  library(Rfastp)
  library(epiwraps)
  library(GEOquery)
  library(GenomicRanges)
  library(IRanges)
  library(ChIPpeakAnno)
  library(ggplot2)
  library(gridExtra)

})

```


```{r, include=FALSE}

tracks_ES_CTCF<- rtracklayer::import("ES/tracks/GSE96107_ES_CTCF.bw")

tracks_ES_H3K27ac<- rtracklayer::import('ES/tracks/ENCFF163HEV.bigWig')

tracks_ES_H3K9me3<- rtracklayer::import('ES/tracks/ENCFF549VBE.bigWig')

peaks_ES_CTCF<- rtracklayer::import('ES/peaks/GSE96107_ES_CTCF.IDR0.05.filt.narrowPeak')


#download.file("https://www.encodeproject.org/files/ENCFF909QOC/@@download/ENCFF909QOC.bed.gz" , dest="ES/peaks/ES_H3K9me3.narrowPeak")

#download.file("https://www.encodeproject.org/files/ENCFF519QMV/@@download/ENCFF519QMV.bed.gz" , dest="ES/peaks/ES_H3K27ac.narrowPeak")

peaks_ES_H3K27ac <- rtracklayer::import('ES/peaks/ES_H3K27ac.narrowPeak')

peaks_ES_H3K9me3<- rtracklayer::import('ES/peaks/ES_H3K9me3.narrowPeak')

```



```{r, collapse=FALSE}
library(Gviz)

ideogram<- IdeogramTrack("chr15", "mm10")
axis<- GenomeAxisTrack()

cover_track_CTCF_ES<-DataTrack(tracks_ES_CTCF, window= 100000,type= "h", name="CTCF", col="blue",ylim = c(0,13))
cover_track_H3K27ac_ES<-DataTrack(tracks_ES_H3K27ac, window= 100000,type= "h", name="H3K27ac", col="purple", ylim = c(0,13))
cover_track_H3K9me3_ES<-DataTrack(tracks_ES_H3K9me3, window= 100000,type= "h", name="H3K9me3", col="pink", ylim = c(0,13) )

head(peaks_ES_CTCF)
peaks_chr15<- peaks_ES_CTCF[seqnames(peaks_ES_CTCF)=="chr15"]
head(peaks_chr15)

peak_track_CTCF_ES<-AnnotationTrack(peaks_chr15,name="Peaks CTCF")
peak_track_H3K27ac_ES<-AnnotationTrack(peaks_ES_H3K27ac,name="Peaks H3K27ac")

peaks_chr15_H3K9me3<- peaks_ES_H3K9me3[seqnames(peaks_ES_H3K9me3)=="chr15"]
peak_track_H3K9me3_ES<-AnnotationTrack(peaks_chr15_H3K9me3,name="Peaks H3K9me3")

library("TxDb.Mmusculus.UCSC.mm10.knownGene")
tx<-GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene ,chromosome = "chr15",start  = 74600000, end= 74690000, name="Genes")

tracks_list_ES<-list(ideogram, cover_track_CTCF_ES, peak_track_CTCF_ES, cover_track_H3K27ac_ES, peak_track_H3K27ac_ES, cover_track_H3K9me3_ES, peak_track_H3K9me3_ES, tx, axis)
```


```{r ChipSeq around Arc in Embrionic Cells, fig.cap="**Figure x:** ChipSeq around Arc in Embrionic Cells, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}


plotTracks(tracks_list_ES,from = 74600000, to= 74690000, chromosome = "chr15",main= "ChipSeq around Arc in Embrionic Stem Cells ",cex.main=1.5)
```
