---
title: "Gadd45y"
author: "Viviana"
date: '2022-07-01'
output: html_document
editor_options: 
  chunk_output_type: console
---

#Analysis of CTCF binding in Gadd45y in Cortical Neurons 

```{r}
#BiocManager::install(c("GEOquery"))
suppressPackageStartupMessages({
  library(AnnotationHub)
  library(Rsubread)
  library(rtracklayer)
  library(Biostrings)
  library(Rfastp)
  library(epiwraps)
  library(GEOquery)
  library(AnnotationHub)
  library(Rsubread)
  library(rtracklayer)
  library(Biostrings)
  library(Rfastp)
  library(epiwraps)
})

library(ChIPpeakAnno)


```


```{r}
library(GenomicRanges)
library(rtracklayer)
library(IRanges)

#Import tracks CN 

tracks_CN_CTCF<- rtracklayer::import('/Users/vivianavinci/Library/Mobile Documents/com~apple~CloudDocs/Desktop/_ETH/SP 2022/BIOINFORMATICS/Project /CN/tracks/GSE96107_CN_CTCF.bw')
```


```{r}
peaks_CN_CTCF<- rtracklayer::import('/Users/vivianavinci/Library/Mobile Documents/com~apple~CloudDocs/Desktop/_ETH/SP 2022/BIOINFORMATICS/Project /CN/peaks/GSE96107_CN_CTCF.IDR0.05.filt.narrowPeak')
```


```{r}
length(peaks_CN_CTCF)

library(Gviz)

ideogram<- IdeogramTrack("chr13", "mm10")
axis<- GenomeAxisTrack()

cover_track_CTCF_CN<-DataTrack(tracks_CN_CTCF, window= 100000,type= "h", name="Coverage")

peak_track_CTCF_CN<-AnnotationTrack(peaks_CN_CTCF,name="Peaks")

library("TxDb.Mmusculus.UCSC.mm10.knownGene")
tx<-GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene, chromosome = "chr13",start  = 51799965, end= 51859475, name="Genes")

plotTracks(list(ideogram,cover_track_CTCF_CN, peak_track_CTCF_CN,tx, axis),from = 51799965, to= 51859475, chromosome = "chr13",main= "CTCF binding around Gadd45 in CN", cex.main=1.5)
```


```{r}

#to be figured out 
count_peaks_NPC_CTCF<-nrow(peaks_NPC_CTCF)

dba.plotHeatmap(peaks_NPC_CTCF, correlations = FALSE)
```

```{r}
sessionInfo()
```

