---
title: "CN_Gadd45"
author: "Viviana"
date: '2022-07-01'
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r, include=FALSE}
suppressPackageStartupMessages({
  library(AnnotationHub)
  library(Rsubread)
  library(rtracklayer)
  library(Biostrings)
  library(Rfastp)
  library(epiwraps)
  library(GEOquery)
  library(GenomicRanges)
  library(IRanges)
  library(ChIPpeakAnno)
  library(ggplot2)
  library(gridExtra)

})

```


```{r, include=FALSE}

tracks_CN_CTCF<- rtracklayer::import('/Users/vivianavinci/Library/Mobile Documents/com~apple~CloudDocs/Desktop/_ETH/SP 2022/BIOINFORMATICS/Project /CN/tracks/GSE96107_CN_CTCF.bw')

tracks_CN_H3K27ac<-rtracklayer::import('CN/tracks/GSE96107_CN_H3K27ac.bw')

tracks_CN_H3K9me3<-rtracklayer::import('CN/tracks/GSE96107_CN_H3K9me3.bw')

peaks_CN_CTCF<- rtracklayer::import('/Users/vivianavinci/Library/Mobile Documents/com~apple~CloudDocs/Desktop/_ETH/SP 2022/BIOINFORMATICS/Project /CN/peaks/GSE96107_CN_CTCF.IDR0.05.filt.narrowPeak')


peaks_CN_H3K27ac <- rtracklayer::import('CN/peaks/GSE96107_CN_H3K27ac.IDR0.05.filt.narrowPeak')

#download.file("https://www.encodeproject.org/files/ENCFF356LFX/@@download/ENCFF356LFX.bed.gz" , dest="CN/peaks/CN_H3K9me3.bed.gz")
peaks_CN_H3K9me3<- rtracklayer::import('CN/peaks/CN_H3K9me3.bed.gz')

```



```{r, collapse=TRUE, warning=FALSE}
library(Gviz)

ideogram<- IdeogramTrack("chr13", "mm10")
axis<- GenomeAxisTrack()

cover_track_CTCF_CN<-DataTrack(tracks_CN_CTCF, window= 100000,type= "h", name="CTCF", col="blue")
cover_track_H3K27ac_CN<-DataTrack(tracks_CN_H3K27ac, window= 100000,type= "h", name="H3K27ac", col="purple")
cover_track_H3K9me3_CN<-DataTrack(tracks_CN_H3K9me3, window= 100000,type= "h", name="H3K9me3", col="pink", ylim = c(0,2) )

peak_track_CTCF_CN<-AnnotationTrack(peaks_CN_CTCF,name="Peaks")
peak_track_H3K27ac_CN<-AnnotationTrack(peaks_CN_H3K27ac,name="Peaks")
peak_track_H3K9me3_CN<-AnnotationTrack(peaks_CN_H3K9me3,name="Peaks")

tx_13<-GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene ,chromosome = "chr13",start  = 51700000, end= 51900000, name="Genes")

tracks_list_CN<-list(ideogram,cover_track_CTCF_CN,peak_track_CTCF_CN, cover_track_H3K27ac_CN, peak_track_H3K27ac_CN, cover_track_H3K9me3_CN,peak_track_H3K9me3_CN, tx_13, axis)
```


```{r, fig.cap="**Figure x:** ChipSeq around Gadd45y in Cortical Neurons, CTCF coverage tracks(blue), H3K27ac coverage tracks(light blue), H3K9me3 coverage tracks (dark blue)"}

plotTracks(tracks_list_CN,from = 51700000, to= 51900000, chromosome = "chr13",main= "ChipSeq around Gadd45y in cortical Neurons",cex.main=1.5)
```


```{r}
# Profile of Chip Seq Binding to Promoters 
library(ChIPseeker)
promoter <- getPromoters(TxDb=TxDb.Mmusculus.UCSC.mm10.knownGene, upstream=3000, downstream=3000)
#Matrix of Promoter and Peaks of CTCF 
tagMatrix <- getTagMatrix(peaks_CN_CTCF, windows=promoter)
##
## to speed up the compilation of this vignettes, we use a precalculated tagMatrix
data("tagMatrixList")
tagMatrix <- tagMatrixList[[4]]
tagHeatmap(tagMatrix, xlim=c(-3000, 3000), color="red",title = "Heatmap of ChIP CTCF peaks binding to TSS regions")

library(org.Mm.eg.db)
peakAnno <- annotatePeak(peaks_CN_CTCF, tssRegion=c(-3000, 3000), TxDb= TxDb.Mmusculus.UCSC.mm10.knownGene , annoDb="org.Mm.eg.db")

Pie_CN_CTCF<- plotAnnoPie(peakAnno)

plotDistToTSS(peakAnno,title="Distribution of transcription factor-binding loci\nrelative to TSS")


peakAnnoList <- lapply(list(peaks_CN_CTCF, peaks_CN_H3K27ac, peaks_CN_H3K9me3) ,annotatePeak, TxDb=TxDb.Mmusculus.UCSC.mm10.knownGene, tssRegion=c(-3000, 3000), verbose=FALSE)
names(peakAnnoList) <- c("CTCF","H3K27ac", "H3K9me3")
plotAnnoBar(peakAnnoList)
plotDistToTSS(peakAnnoList)

```


```{r}
length(peaks_CN_CTCF)
head(peaks_CN_CTCF)

#peakSignals_CTCF<- signal2Matrix('CN/tracks/GSE96107_CN_CTCF.bw', regions=peaks_CN_CTCF, extend=3000, w=20)

#Coverage of H3K27ac around CTCF binding in CN 
peaksSignals_H3K27ac<-signal2Matrix('CN/tracks/GSE96107_CN_H3K27ac.bw', regions = peaks_CN_CTCF, extend = 3000, w = 20 )
peaksSignals_H3K9me3<-signal2Matrix('CN/tracks/GSE96107_CN_H3K9me3.bw', regions = peaks_CN_CTCF, extend = 3000, w = 20 )

#Heatmap of H3K9me3 coverage around CTCF peaks in CN 
heatmapH3K27ac<- plotEnrichedHeatmaps(peaksSignals_H3K27ac, trim=0.9, scale_title = "density 1")
heatmapH3K9me3<- plotEnrichedHeatmaps(peaksSignals_H3K9me3, trim=0.9, scale_title = "density 2")





grid.newpage()
pushViewport(viewport(layout = grid.layout(nr = 1, nc = 2)))
pushViewport(viewport(layout.pos.row = 1, layout.pos.col = 1))
draw(heatmapH3K27ac, column_title = "H3K27ac around CTCF", show_heatmap_legend = FALSE, newpage = FALSE)
upViewport()

pushViewport(viewport(layout.pos.row = 1, layout.pos.col = 2))
draw(heatmapH3K9me3, column_title = "H3K9me3 around CTCF", show_heatmap_legend = FALSE, newpage = FALSE)
upViewport()



pushViewport(viewport(layout.pos.row = 1, layout.pos.col = 2))
upViewport()



```

