---
author: "Pierre-Luc"
date: "3/31/2022"
output: html_document
---

```{r}
suppressPackageStartupMessages({
  library(GenomicRanges)
  library(memes)
  library(Biostrings)
  library(AnnotationHub)
  library(MotifDb)
  library(universalmotif)
  library(PWMEnrich)
  library(ggplot2)
})
```

# Obtaining peak sequences

For the purpose of this example, we'll use the CTCF peaks found on chromosome 1 of mESC:

```{r}
download.file("https://www.encodeproject.org/files/ENCFF508CKL/@@download/ENCFF508CKL.bed.gz", "mESC_CTCF_ENCFF508CKL.bed.gz")
peaks <- rtracklayer::import("mESC_CTCF_ENCFF508CKL.bed.gz", format="NarrowPeak")
seqlevelsStyle(peaks) <- "Ensembl"
peaks_chr1 <- peaks[seqnames(peaks)=="1"]
```


Since the motif of a transcription factor is typically found towards the center or summit of the peak, it is common practice to look for motifs around the center of peaks. 
How far around depends on the resolution of your data, i.e. how narrow are your peaks (can be between 50bp to 500bp around).
The first thing we do, then, is to extract the regions around the peaks, and then obtain the sequence from those regions:

```{r}
peak_centers <- resize(peaks_chr1, fix="center", width=100)
# we get the genome sequence:
ah <- AnnotationHub(localHub=TRUE)
genome <- ah[["AH68356"]]
# we get the sequences corresponding to the peak centers:
peak_seqs <- memes::get_sequence(peak_centers, genome)
peak_seqs
```


# Motif discovery

Motif discovery aims at finding new motifs that are enriched in a set of sequences (e.g. peaks) versus a background (e.g. random or control sequences).

## Example using Meme

```{r}
# this requires having the meme suite installed
mr <- runMeme(peak_seqs,
              p=4, # use 4 threads
              nmotifs=20, # determines how many motifs Meme will try to find
              meme_path = "/home/pierre-luc/bin/") # depending on your setup, you might need to specify where the meme binaries were installed
```

You could also specify a background using the `neg` argument, e.g.
```{r, eval=FALSE}
mr <- runMeme(peak_seqs,
              neg=non_peaks, # instead of looking for motifs against a random background, you can specify control (i.e. non-peak) sequences
              nmotifs=20)
```


## Example using rGADEM

```{r}
gadem <- rGADEM::GADEM(peak_seqs, genome=genome)
```






# Motif scanning

Motif scanning aims at finding the _occurrences_ of a known motif in a set of sequences.
As this is methodologically fairly simple, it doesn't really matter what method one uses, though results may differ because of how the thresholds are set.

## Getting the desired motif

```{r}
# we search for "CTCF" in the motif database
motifs <- query(MotifDb, "CTCF")
# there are several matching motifs:
names(motifs)
# we select one:
motif <- motifs[["Hsapiens-JASPAR_CORE-CTCF-MA0139.1"]]
# we visualize it:
view_motifs(motif)
```


## Scanning a set of sequences using the Meme suite

```{r}
moi <- memes::runFimo(peak_seqs,  convert_motifs(motif))
# you might eventually need to add the meme_path argument:
moi <- memes::runFimo(peak_seqs,  convert_motifs(motif), meme_path = "/home/pierre-luc/bin/")
```


## Alternative (slower, but doesn't require Meme installation):

You'll first need this wrapper function:

```{r}
#' findMotifInstances
#'
#' A wrapper around TFBSTools for scanning motif occurence, handling the 
#' coordinate conversion as `memes` does.
#' 
#' @param seqs A set of sequences, e.g. `DNAStringSet`, optionally with 
#'   coordinate as names to enable conversion.
#' @param motif A motif, in any format recognized by `universalmotif`
#' @param keepMatchedSeq Logical; whether to keep the matched sequence.
#' @param ... Passed to `TFBSTools::searchSeq`; can for instance be used to set
#'   the number of threads to use, e.g. with `mc.cores=2`
#'
#' @return A `GRanges` object
findMotifInstances <- function(seqs, motif, keepMatchedSeq=FALSE, ...){
  motif <- universalmotif::convert_motifs(motif, "TFBSTools-PWMatrix")
  x <- suppressWarnings(TFBSTools::searchSeq(motif, subject=seqs))
  peaks <- strsplit(gsub("-",":",names(seqs)), ":")
  if(all(lengths(peaks)==3)){ # convert relative coordinates to absolute
    chrs <- sapply(peaks,FUN=function(x) x[1])
    offsets <- sapply(peaks,FUN=function(x) as.integer(x[2]))-1L
    i <- rep(seq_along(x),lengths(x))
    return(GRanges(chrs[i], 
                   IRanges(offsets[i]+as.integer(unlist(lapply(x, start))),
                           offsets[i]+as.integer(unlist(lapply(x, end)))),
                   strand=unlist(lapply(x, strand)),
                   score=as.numeric(unlist(lapply(x, FUN=function(x) score(x))))
                   ))
  }
  x <- as(x, "GRanges")
  score(x) <- x$absScore
  keepFields <- c("score","relScore")
  if(keepMatchedSeq) keepFields <- c(keepFields, "siteSeqs")
  mcols(x) <- mcols(x)[,intersect(colnames(mcols(x)), keepFields)]
  x
}
```


Then you can get the motif instances like this:

```{r}
moi2 <- findMotifInstances(peak_seqs, motif, mc.cores=2) # running with 2 threads
```






# Motif enrichment analysis

Motif enrichment analysis aims at finding _known_ motifs that are enriched in a set of sequences (e.g. peaks) versus a background.
By default, the background is generated by shuffling the input sequences, but it can also be specified.

## Example using AME from the Meme suite

```{r}
# obtain the motifs (e.g. here use all the mouse motifs)
motifs <- query(MotifDb, c("Mmusculus"))
ame <- runAme(peak_seqs, database=convert_motifs(motifs), meme_path="/home/pierre-luc/bin/")
head(ame)
```

We could also plot the results:

```{r}
ggplot(ame, aes(log2((1+tp)/(1+fp)), -log10(adj.pvalue), size=tp_percent)) + 
  geom_point(alpha=0.3) + geom_text(aes(label=motif_id), data=head(ame)) +
  labs(x="log2(fold-enrichment)", size="% of set")
```


## Example using PWMEnrich (much slower)

```{r}
# obtain the motifs (e.g. here use the core JASPAR mouse motifs)
motifs <- query(MotifDb, c("JASPAR_CORE", "Mmusculus"))
# prepare multithreading:
registerCoresPWMEnrich(4)
# launch the analysis:
res <- motifEnrichment(peak_seqs, convert_motifs(motifs, "PWMEnrich-PWM"))
# for the sake of example, we'll just use the first 300 sequences:
res <- motifEnrichment(head(peak_seqs, convert_motifs(motifs, "PWMEnrich-PWM"))
groupReport(res)
```
