---
title: "NPC_CTCF_Gadd45"
author: "Viviana"
date: '2022-07-01'
output: html_document
editor_options: 
  chunk_output_type: console
---


#Analysis of CTCF binding in Gadd45y in Neuronal Progenitor Neurons 

```{r}

suppressPackageStartupMessages({
  library(AnnotationHub)
  library(Rsubread)
  library(rtracklayer)
  library(Biostrings)
  library(Rfastp)
  library(epiwraps)
  library(GEOquery)
  library(AnnotationHub)
  library(Rsubread)
  library(rtracklayer)
  library(Biostrings)
  library(Rfastp)
  library(epiwraps)
  library(GenomicRanges)
  library(rtracklayer)
  library(IRanges)
  library(ChIPpeakAnno)
  library(ggplot2)
  library(gridExtra)

})



```


```{r}
getwd()

tracks_NPC_CTCF<- rtracklayer::import("NPC /tracks/GSE96107_NPC_CTCF.bw")
```


```{r}
peaks_NPC_CTCF<- rtracklayer::import('NPC /peaks/GSE96107_NPC_CTCF.IDR0.05.filt.narrowPeak')
```


```{r} 

#just Playing with Overlaps 
#length(peaks_NPC_CTCF)
#overlap_NPC_ES<-overlapsAny(peaks_NPC_CTCF,peaks_ES_CTCF)
#table(overlap_NPC_ES)
#Overlapping<- subsetByOverlaps (peaks_NPC_CTCF,peaks_ES_CTCF)
#length(Overlapping)
```


```{r}


library(Gviz)

ideogram<- IdeogramTrack("chr13", "mm10")
axis<- GenomeAxisTrack()

cover_track_CTCF_NPC<-DataTrack(tracks_NPC_CTCF, window= 100000,type= "h", name="Coverage")

peak_track_CTCF_NPC<-AnnotationTrack(peaks_NPC_CTCF,name="Peaks")

library("TxDb.Mmusculus.UCSC.mm10.knownGene")
tx<-GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene, chromosome = "chr13",start  = 51799965, end= 51859475, name="Genes")

plotTracks(list(ideogram,cover_track_CTCF_NPC, peak_track_CTCF_NPC,tx, axis),from = 51799965, to= 51859475, chromosome = "chr13",main= "CTCF binding around Gadd45 in NPC", cex.main=1.5)
```


```{r}

#to be figured out 
#count_peaks_NPC_CTCF<-nrow(peaks_NPC_CTCF)

#dba.plotHeatmap(peaks_CN_CTCF, correlations = FALSE)
```

```{r}
sessionInfo()
```

